workflow/WorkflowService.go

```diff
@@ -142,6 +142,75 @@ func (service *WorkflowService) Init(
 
 	service.cron = cron.New()
 
+	// allow the database time to settle
+	//time.Sleep(time.Second * 5)
+	pollingFn := elasticsearch.GetData(func() (bool, error) {
+		exists:=eventtypesIndex.IndexExists()
+		types, err := eventtypesIndex.GetTypes()
+		if err != nil {
+			//handle error
+			fmt.Println("error getting types")
+		}
+		fmt.Printf("Getting %d types...\n", len(types))
+		if len(types) == 0 {
+			return false, nil
+		}
+		for _, typ := range types {
+			fmt.Println(typ)
+		}
+		fmt.Printf("Exists: %t\n", exists)
+		return exists, nil
+	})
+
+	pollOk, pollErr := elasticsearch.PollFunction(pollingFn)
+	if pollErr != nil {
+		fmt.Print("ERROR", pollErr, "\n")
+	}
+	fmt.Print("SETUP INDEX\n", pollOk, "\n")
+
+	// Ingest event type
+	ingestEventType := &EventType{}
+	ingestEventType.Name = "piazza:ingest"
+	ingestEventTypeMapping := map[string]elasticsearch.MappingElementTypeName{
+		"dataId":   "string",
+		"dataType": "string",
+		"epsg":     "short",
+		"minX":     "long",
+		"minY":     "long",
+		"maxX":     "long",
+		"maxY":     "long",
+		"hosted":   "boolean",
+	}
+	ingestEventType.Mapping = ingestEventTypeMapping
+	log.Println("  Creating piazza:ingest eventtype")
+	postedIngestEventType := service.PostEventType(ingestEventType)
+	log.Printf("  Created piazza:ingest eventtype: %s", postedIngestEventType.StatusCode)
+	if postedIngestEventType.StatusCode == 201 {
+		// everything is ok
+		//service.logger.Info("Created EventType with EventTypeId %s", postedIngestEventType.Data)
+		log.Printf("  SUCCESS Created piazza:ingest eventtype: %s", postedIngestEventType.StatusCode)
+	} else {
+		// something is wrong
+		//service.logger.Info("Problem creating EventType %s", postedIngestEventType.StatusCode)
+		log.Printf("  ERROR creating piazza:ingest eventtype: %s", postedIngestEventType.StatusCode)
+	}
+
+	// Execution Completed event type
+	executionCompletedType := &EventType{}
+	executionCompletedType.Name = "piazza:executionComplete"
+	executionCompletedTypeMapping := map[string]elasticsearch.MappingElementTypeName{
+		"jobId":    "string",
+		"status":   "string",
+		"dataId":   "string",
+	}
+	executionCompletedType.Mapping = executionCompletedTypeMapping
+	postedExecutionCompletedType := service.PostEventType(executionCompletedType)
+	if postedExecutionCompletedType.StatusCode == 201 {
+		// everything is ok
+	} else {
+		// something is wrong or it was already there
```

Comment:
Do we want to have a more explicit failure here? Or at least some logging like the previous check.